common/lockmanager_test.go

```diff
@@ -0,0 +1,114 @@
+package common
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edulinq/autograder/config"
+)
+
+var (
+    key = "testkey1"
+    key2 = "testkey2"
+    doesNotExistKey = "dne"
+    staleDuration = time.Duration(config.STALELOCK_DURATION.Get()) * time.Second;
+    wg sync.WaitGroup;
+)
+
+func TestLockManager(test *testing.T) {
+    testLockingKey()
+    
+    testUnlockingKey(test)
+
+    testConcurrentLockingUnlocking()
+
+    testUnlockingAnUnlockedLock(test)
+    
+    testUnlockingMissingKey(test)
+
+    testLockConcurrencyWithStaleCheck(test)
+}
+
+func testLockingKey() {
+    Lock(key)
+}
+
+func testUnlockingKey(test *testing.T) {
```

Comment:
There are two issues with this test and `testLockingKey()`:
 - They are dependent (each test should be independent).
 - They are too simple.

The last one may seem strange (usually we want simple),
but the problem here is that they are simple to the point of being incomplete.
Instead, just have a single test (`TestLockBase()`?) that just locks and unlocks.